pkg/cloud/cloud.go

```diff
@@ -498,6 +498,104 @@ func (c *cloud) ModifyDisk(ctx context.Context, volumeID string, options *Modify
 	return c.waitForVolumeSize(ctx, volumeID)
 }
 
+func (c *cloud) ResizeOrModifyDisk(ctx context.Context, volumeID string, newSizeBytes int64, options *ModifyDiskOptions) (int64, error) {
```

Comment:
Suggest breaking up the logic for fetching volume information, building the `ModifyVolumeInput` req, handling the response, and checking/returning the final size into separate functions.

Reply:
Added a separate function ``fetchVolumeInformation``to make this more organized